File: 7-kyu/sum-of-a-sequence/sum-of-a-sequence.py
def sequence_sum(begin, end, step):
    if begin > end:
        return 0
    if (end - begin) % step != 0:
        end = begin + step * ((end - begin) // step)
    num = ((end - begin) // step) + 1
    total_sum = (num / 2) * (begin + end)
    return total_sum
​

File: 7-kyu/sum-of-a-sequence/sum-of-a-sequence.go
package kata
​
​
func SequenceSum(start, end, step int) int {
  num := ((end - start)/step) + 1
  sum := (num * (start + end))/2
  if end < start{
    return 0
    } else if (end - start) % step != 0 {
        end1 := start + step * ((end-start)/step)
          return (num * (start + end1))/2
  }
  return sum
}
​